src/charm_config.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Config of the Charm."""
+
+import dataclasses
+import logging
+from enum import Enum
+from ipaddress import IPv4Network, IPv6Network
+from typing import Optional
+
+import ops
+from pydantic import (  # pylint: disable=no-name-in-module,import-error
+    BaseModel,
+    ConfigDict,
+    Field,
+    StrictStr,
+    ValidationError,
+)
+from pydantic.functional_validators import model_validator
+from pydantic.networks import IPvAnyAddress, IPvAnyNetwork
+from pydantic_core import PydanticCustomError
+from pydantic_extra_types.mac_address import MacAddress
+from typing_extensions import TypeAlias
+
+logger = logging.getLogger(__name__)
+
+
+class CNIType(str, Enum):
+    """Class to define available CNI types for UPF operator."""
+
+    bridge = "bridge"
+    macvlan = "macvlan"
+    host_device = "host-device"
+    vfioveth = "vfioveth"
+
+
+class UpfMode(str, Enum):
+    """Class to define available UPF modes for UPF operator."""
+
+    af_packet = "af_packet"
+    dpdk = "dpdk"
+
+
+NetworkType: TypeAlias = "str | bytes | int | tuple[str | bytes | int, str | int]"
+
+
+class LaxIPvAnyNetwork(IPvAnyNetwork):
```

Comment:
The class name did not make sense at the first glance.

Reply:
I used `Lax` as opposite of `Strict`. I can think of `Loose`. Do you have any suggestion?

Reply:
NonStrict ? I was also confused about the name